internal/provider/validators.go

```diff
@@ -106,3 +108,39 @@ func validateDatacenter(v interface{}, path cty.Path) diag.Diagnostics {
 
 	return diagnostics
 }
+
+func validateConsulClusterTier(v interface{}, path cty.Path) diag.Diagnostics {
+	var diagnostics diag.Diagnostics
+
+	err := consulmodels.HashicorpCloudConsul20210204ClusterConfigTier(strings.ToUpper(v.(string))).Validate(strfmt.Default)
+	if err != nil {
+		expectedEnumList := regexp.MustCompile(`\[.*\]`).FindStringSubmatch(err.Error())
+		msg := fmt.Sprintf("expected %v to be one of %v", v, expectedEnumList[0])
```

Comment:
A little confusing that `expectedEnumList` seems to be a list of lists 🤔 Any way to flatten?